lib/uniqueBy.js

```diff
@@ -0,0 +1,58 @@
+const assert = require( 'assert' );
+const isFunction = require( 'lodash/isFunction' );
+const isString = require( 'lodash/isString' );
+const get = require( 'lodash/get' );
+
+const DuplicateEventKey = 'rejected';
+
+/**
+ * Emits only unique inputs based on the given attributeName via the 'data' event
+ * and through the pipe. Duplicated items are emitted via the 'duplicate' event.
+ *
+ * @param {IterateeFunction} iteratee
+ *  The iteratee function invoked per element
+ *
+ * @constant {String} DuplicateEventKey
+ *  The key of the event emitted when duplicate items are found.
+ *
+ * @returns {object}
+ *  returns a Transform stream that applies the provided iteratee to emit only unique items
+ *
+ * @callback IterateeFunction
+ * @param {any} item
+ *  input parameter for the iteratee function to return the value that will be evaluated to be unique
+ * @returns {any}
+ */
+
+function createUniqueByStream( iteratee ){
+
+  assert( isFunction( iteratee ) || isString( iteratee ), 'Iteratee can be eitehr a function or a string' );
+
+  const cache = new Set();
+
+  function transform( item, _, next ){
+    
+
+    const v = isFunction( iteratee )
```

Comment:
look at `https://lodash.com/docs/4.17.15#iteratee` and used it instead of the assert above